File: modules/wem-api/src/main/java/com/enonic/wem/api/schema/content/ContentTypeUpdatedEvent.java
package com.enonic.wem.api.schema.content;

import java.time.Instant;

import com.enonic.wem.api.event.Event;

public class ContentTypeUpdatedEvent
    implements Event
{
    private final ContentTypeName name;

    private final Instant modifiedTime;

    public ContentTypeUpdatedEvent( final ContentTypeName name, final Instant modifiedTime )
    {
        this.name = name;
        this.modifiedTime = modifiedTime;
    }

    public ContentTypeName getName()
    {
        return name;
    }

    public Instant getModifiedTime()
    {
        return modifiedTime;
    }
}

File: modules/admin-ui/src/main/resources/web/admin/common/js/schema/content/ContentTypeUpdatedEvent.ts
module api.schema.content {

    export class ContentTypeUpdatedEvent extends api.event.Event {

        private contentTypeName: ContentTypeName;

        private modifiedTime: Date;

        constructor(name: ContentTypeName, modifiedTime: Date) {
            super();
            this.contentTypeName = name;
            this.modifiedTime = modifiedTime;
        }

        getContentTypeName(): ContentTypeName {
            return this.contentTypeName;
        }

        getModifiedTime(): Date {
            return this.modifiedTime;
        }

        static on(handler: (event: ContentTypeUpdatedEvent) => void) {
            api.event.Event.bind(api.ClassHelper.getFullName(this), handler);
        }

        static un(handler?: (event: ContentTypeUpdatedEvent) => void) {
            api.event.Event.unbind(api.ClassHelper.getFullName(this), handler);
        }


        static fromJson(json: ContentTypeUpdatedEventJson): ContentTypeUpdatedEvent {
            var modifiedTime = json.modifiedTime ? new Date(Date.parse(json.modifiedTime)) : null;
            return new ContentTypeUpdatedEvent(new ContentTypeName(json.name), modifiedTime);
        }
    }

    export interface ContentTypeUpdatedEventJson {

        name: string;

        modifiedTime: string;
    }
}